File: src/Sorting/SelectionSort/selection_sort.py
def selection_sort(array):
    for i in range(len(array)):
        smallest = find_index_of_smallest_element(array, i)
        array[i], array[smallest] = array[smallest], array[i]


def find_index_of_smallest_element(array, initial_index):
    smallest_index = initial_index
    for i in range(initial_index, len(array)):
        if array[smallest_index] > array[i]:
            smallest_index = i

    return smallest_index

File: src/Sorting/SelectionSort/SelectionSort.ts
/**
 * Sorts array in place with selection sort algorithm.
 *
 * @param {number[]} array
 */
export function selectionSort(array: number[]) {
  for (let i = 0; i < array.length; i++) {
    const smallest = findIndexOfSmallestElement(array, i);
    [array[i], array[smallest]] = [array[smallest], array[i]];
  }
}

/**
 * Finds the smallest element in array starting from initialIndex.
 *
 * @param {number[]} array
 * @param {number} initialIndex - starting position
 * @return {number} index of the smallest element in a given array.
 */
function findIndexOfSmallestElement(array: number[], initialIndex: number) {
  let smallestIndex = initialIndex;

  for (let i = initialIndex + 1; i < array.length; i++) {
    if (array[i] < array[smallestIndex]) {
      smallestIndex = i;
    }
  }

  return smallestIndex;
}
